src/docfx/cli/New.cs

```diff
@@ -83,7 +83,15 @@ private static bool CreateFromTemplate(string templateName, CommandLineOptions o
             }
             else
             {
-                Directory.CreateDirectory(Path.GetDirectoryName(targetFullPath) ?? ".");
+                var directory = Path.GetDirectoryName(targetFullPath) ?? ".";
+                if (!Directory.Exists(directory))
+                {
+                    Directory.CreateDirectory(directory);
+                    if (TestQuirks.Verbose ?? false)
```

Comment:
It seems weird to initialize a folder to Git when `Verbose == true`.

Reply:
It's because we want to scan all redirection files within the repo folder, the `_buildOptions.repopath`  is from [here (link)](https://github.com/dotnet/docfx/blob/6163e1c90c8f316789051d511bb1f8a9636ecae5/src/docfx/lib/git/GitUtility.cs#L17), and for these tests without any repo, it will traverse upwards until find a git directory.

For these tests, it will find docfx itself is a git directory, and then we will get all redirection files in all spec tests, which will fail lots tests. So I just want to make sure the git directory is limited within the test folder.

![image](https://user-images.githubusercontent.com/67643974/181202119-c39de795-48c4-4055-9a3d-5b7a84314bfb.png)

Reply:
If so, would it be better to use a separate flag?

Reply:
Done